pkg/plugin/trivy/plugin.go

```diff
@@ -839,31 +900,58 @@ func (p *plugin) getPodSpecForClientServerMode(ctx starboard.PluginContext, conf
 			},
 		}
 
-		if _, ok := credentials[container.Name]; ok && secret != nil {
-			registryUsernameKey := fmt.Sprintf("%s.username", container.Name)
-			registryPasswordKey := fmt.Sprintf("%s.password", container.Name)
+		TTLResult, err := config.GetECRRefreshTTL()
+		if err != nil {
+			TTLResult = "0"
+		}
+
+		if config.UseECRCredentials() && CheckAwsEcrPrivateRegistry(container.Image) != "" {
+
+			if utils.TokenTTLValidation(EcrTokenGen, TTLResult) {
+				EcrTokenGen = time.Now()
+				aws_creds, err = GetAuthorizationToken(CheckAwsEcrPrivateRegistry(container.Image))
+				if err != nil {
+					return corev1.PodSpec{}, nil, err
```

Comment:
`EcrTokenGen` should be set to `time.Now()` only if the `GetAuthorizationToken` returns no error.  to avoid `index out of range error` in the next authorization cycle as `aws_creds` will be nil

Reply:
Agreed and fixed.